quic/s2n-quic-platform/src/io/tokio/builder.rs

```diff
@@ -108,7 +109,26 @@ impl Builder {
 
     /// Sets the largest maximum transmission unit (MTU) that can be sent on a path
     pub fn with_max_mtu(mut self, max_mtu: u16) -> io::Result<Self> {
-        self.max_mtu = max_mtu
+        self.mtu_config.max_mtu = max_mtu
+            .try_into()
+            .map_err(|err| io::Error::new(ErrorKind::InvalidInput, format!("{err}")))?;
+        Ok(self)
+    }
+
+    /// Sets the maximum transmission unit (MTU) to use when initiating a connection
```

Comment:
Can we expand the documentation here a bit more? When would you want to each one of these options? Add some details around invariants (e.g. initial_mtu <= max_mtu). We should also document what we mean by MTU :) since it can quickly become ambiguous.

Reply:
I added some much more detailed docs for each of these

Reply:
Yeah those look great. Thanks for doing that!